### PR TITLE
Use JSON logging in production

### PR DESCRIPTION
**Pull request by jb3**
JSON logging allows for greater filtering with our log aggregator, Loki.

This PR introduces JSON logging when debug mode is set to false (i.e. the bot is running in production).

Screenshot 2021-03-06 at 12 01 02

### CODE REVIEW

**Review by Shivansh-007 — APPROVED** [review]
Tada, tested locally!

**Review by HassanAbouelela — CHANGES_REQUESTED** [review]
Interestingly, some things slip through the JSON formatting (probably because they aren't logged, just straight up warnings/errors, such as not having `python-Levenshtein` installed), but that's not really important for the bot, just something you'd have to account for on the Loki end.

File: Pipfile [hunk 1]
```diff
@@ -28,6 +28,7 @@ sphinx = "~=2.2"
 statsd = "~=3.3"
 arrow = "~=0.17"
 emoji = "~=0.6"
+python-json-logger = "*"
```
[the remark this answers is not in the document]
Comment (jb3) [reply Pipfile]: Addressed in fe5e2ab947a8c2123f2856db24d9c6e978adec9d

**Review by HassanAbouelela — APPROVED** [review]

**Comment by jb3** [comment]
Hold on, need a Pipfile.lock bump

**Review by HassanAbouelela — APPROVED** [review]
One more approval for old time's sake.

**Comment by HassanAbouelela** [comment]
This PR is being reverted due to an internal decision. See discussion [here](https://discord.com/channels/267624335836053506/411200599653351425/818541370083377152).

**Pull request merged**